corehq/apps/userreports/models.py

```diff
@@ -817,7 +817,7 @@ def by_ids(cls, config_ids):
         mapping = cls.by_id_mapping()
 
         return_configs = []
-        for config_id in set(config_ids):
+        for config_id in config_ids:
```

Comment:
Can't you leave the `set`? The later logic - creating the dict and populating the return value based on that - is enough to fix the bug.

Reply:
having set is redundant here. Its a list that is passed. I don't see why we should make it a set.

Reply:
I'm guessing it's in case there are duplicates in the config_ids passed in, so the loop doesn't run unnecessarily.

Reply:
I guess i was trying to maintain the same order of reports/configs by ids and hence didn't see the obvious. Moving the re-ordering to the calling method post discussion and hence adding this back.